lightning/src/ln/channelmanager.rs

```diff
@@ -836,7 +836,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 
 		let config = if override_config.is_some() { override_config.as_ref().unwrap() } else { &self.default_configuration };
-		let channel = Channel::new_outbound(&self.fee_estimator, &self.keys_manager, their_network_key, channel_value_satoshis, push_msat, user_id, config)?;
+		let block_hash = *self.last_block_hash.lock().unwrap();
+		let channel = Channel::new_outbound(&self.fee_estimator, &self.keys_manager, their_network_key, channel_value_satoshis, push_msat, user_id, config, block_hash)?;
```

Comment:
@TheBlueMatt It occurred to me while working on a follow-up that we may want to hold the `last_block_hash` lock for the duration of creating the channel (i.e., also covering the duration `channel_state` lock is held here). And that the same should be done in `block_connected` / `block_disconnected`.

Otherwise, we may create a channel with a block hash, process a new block, then add the new channel after the block has been processed but with the previous block hash. But maybe this is not a problem?

However, holding locks in this manner means `create_channel` / `internal_open_channel` would block `block_connected` / `block_disconnected`. And this includes calling out to the fee estimator.

(Note that both already involve holding these locks only in a non-overlapping manner.)

Does arranging locking like this sound reasonable? If not clear, I'm proposing:
 - lock `last_block_hash`
 - lock `channel_state`
 - unlock `channel_state`
 - unlock `last_block_hash`

Reply:
Ah! That is a problem as we'll refuse to deserialize a `ChannelManager` that has Channels with different current block hashes set (as long as they aren't 0). I think an easier fix is to just move the last_block_hash update to the top of block_connected/disconnected, which avoids taking locks for longer.

Reply:
Won't channels still get different block hashes in the following scenario?
- `create_channel` locks `last_block_hash` and uses this for the new channel
- `create_channel` unlocks `last_block_hash`
- `block_connected` locks `last_block_hash` and updates its value
- `block_connected` unlocks `last_block_hash`
- `block_connected` locks `channel_state` and calls `block_connected` on each channel
- `block_connected` unlocks `channel_state`
- `create_channel` locks `channel_state` and inserts the new channel but with the previous `last_block_hash`
- `create_channel` unlocks `channel_state`

Another question that came to mine: does it matter that a new `Channel` is given `last_block_hash` even though `block_connected` was not called on it for that hash?

Reply:
> Won't channels still get different block hashes in the following scenario?

Ugh, indeed. Seems fine enough to just hold the lock until the channel is inserted.

> Another question that came to mine: does it matter that a new Channel is given last_block_hash even though block_connected was not called on it for that hash?

I don't believe so. The channel-current-hash is basically unused, its now just used to initialize the ChannelMonitor. Its checked at deserialization time, but that's about it. Maybe an easier way to solve the locking issue is just to remove it.

Reply:
> I don't believe so. The channel-current-hash is basically unused, its now just used to initialize the ChannelMonitor. Its checked at deserialization time, but that's about it. Maybe an easier way to solve the locking issue is just to remove it.

Ah, good idea! So just pass it to `Channel` when needed to create a `ChannelMonitor`.

Reply:
Sounds good to me.